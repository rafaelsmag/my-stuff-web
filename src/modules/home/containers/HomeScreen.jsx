import React, { Component } from 'react'
import { Grid } from 'semantic-ui-react'
import { getPosts } from '../../../services/backend/postService'
import FeedCard from '../components/FeedCard'

class HomeScreen extends Component {
  state = {
    posts: []
  }

  componentDidMount () {
    const posts = getPosts()
    this.setState({
      posts
    })
  }

  renderFeed = () => {
    return this.state.posts.map((post) => {
      return (
        <FeedCard key={post._id} post={post}/>
      )
    })
  }

  render () {
    return (
      <Grid columns={1} centered >
        <Grid.Column mobile={15} tablet={10} computer={8}>
          {this.renderFeed()}
        </Grid.Column>
      </Grid>
    )
  }
}

export default HomeScreen
