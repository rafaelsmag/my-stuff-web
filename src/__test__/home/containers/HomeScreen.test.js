import React from 'react'
import { shallow } from 'enzyme'

import HomeScreen from '../../../modules/home/containers/HomeScreen'

describe('HomeScreen Component', () => {
  const homeScreen = shallow(<HomeScreen />)
  it('should redenders properly', () => {
    expect(homeScreen).toMatchSnapshot()
  })

  it('Initializes the `state` with an empty list of posts', () => {
    expect(homeScreen.state().posts).toEqual([])
  })
})
