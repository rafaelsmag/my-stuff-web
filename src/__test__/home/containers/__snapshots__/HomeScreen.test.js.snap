// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Component should redenders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "centered": true,
      "children": <GridColumn
        computer={8}
        mobile={15}
        tablet={10}
>
        
</GridColumn>,
      "columns": 1,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
        "computer": 8,
        "mobile": 15,
        "tablet": 10,
      },
      "ref": null,
      "rendered": Array [],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "centered": true,
        "children": <GridColumn
          computer={8}
          mobile={15}
          tablet={10}
>
          
</GridColumn>,
        "columns": 1,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [],
          "computer": 8,
          "mobile": 15,
          "tablet": 10,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
